python/cudf/cudf/core/dataframe.py

```diff
@@ -3722,6 +3722,49 @@ def sort_values(
             keep_index=not ignore_index,
         )
 
+    def agg(self, aggs):
+        dtypes = [self[col].dtype for col in self._column_names]
+        common_dtype = np.find_common_type(dtypes, [])
+        df_normalized = self.astype(common_dtype)
+
+        if isinstance(aggs, list):
+            result = cudf.DataFrame()
+            for agg in aggs:
+                result[agg] = getattr(df_normalized, agg)()
+
+        elif isinstance(aggs, str):
+            result = cudf.DataFrame()
+            result[aggs] = getattr(df_normalized, aggs)()
+            result = result.T.loc[aggs]
+            result.name = None
+            return result
+
+        elif isinstance(aggs, dict):
+            cols = aggs.keys()
+            idxs = set()
+            for agg_l in aggs.values():
+                for agg in agg_l:
+                    idxs.add(agg)
+            idxs = sorted(list(idxs))
+            result = cudf.DataFrame(index=idxs, columns=cols)
+            for key in aggs.keys():
+                col = df_normalized[key]
+                ans = cudf.Series(
+                    [None] * len(idxs), index=idxs, dtype=col.dtype
```

Comment:
Instead of initializing the Series via a Python list of `None`, we should use our `column_empty` utility to prevent unnecessay host --> device copies: https://github.com/rapidsai/cudf/blob/branch-0.17/python/cudf/cudf/core/column/column.py#L1318